spring-cloud-gcp-pubsub/src/main/java/com/google/cloud/spring/pubsub/core/PubSubConfiguration.java

```diff
@@ -81,6 +83,23 @@ public Subscriber getSubscriber(String name, String projectId) {
 		return subscriberProperties;
 	}
 
+	/**
+	 * Computes the number of executor threads. The subscription-specific property takes
+	 * precedence if both global and subscription-specific properties are set. If none are set
+	 * then the default (4) is returned.
+	 * @param subscriptionName subscription name
+	 * @param projectId project id
+	 * @return number of executor threads
+	 */
+	public int computeSubscriberExecutorThreads(String subscriptionName, String projectId) {
+		Integer numThreads = getSubscriber(subscriptionName, projectId).getExecutorThreads();
+		if (numThreads != null) {
+			return numThreads;
+		}
+		Integer globalExecutorThreads = subscriber.getExecutorThreads();
```

Comment:
Minor nit: Would be nice if you could rename `subscriber` to `defaultSubscriber` to provide a clearer difference between `subscriber` and `getSubscriber(subscriptionName, projectId)`.

Reply:
That's a good idea! I assumed Spring autoconfig wouldn't be able to pick up the global config if the variable was renamed to something else but interestingly it works when the getter's name stays as `getSubscriber()`.